src/search_engines.js

```diff
@@ -1344,6 +1344,50 @@ class Github {
   }
 }
 
+class Gitlab {
+  constructor(options) {
+    this.options = options;
+  }
+  get urlPattern() {
+    return /^https:\/\/gitlab-ext\.exxcellent\.de/;
```

Comment:
Don't hardcode the Gitlab instance URL, it should be possible for the user to set it in the options

Reply:
So this is where i have a Problem. Is my Implementation in options_page.html correct?

I don't know how I can provide a Regex and the gitlab-url option that is only a string.

I also don't realy like to add this code in the manifest.json: 
```
    "optional_permissions": [
        "https://*/",
```
Do you have an idea to how we can add those options?

Reply:
From a quick look it the implementation seems ok.
Yes, adding `https://*` to the `optional_permissions` is problematic and degrades the extension's security, so we should avoid it. We could initially add only the specific gitlab instances that users actually use, but that seems a bit strange. Not sure what's the best thing to do here

Reply:
Got it to work.

I cleaned up the code except one id definition.
It looks like `yarn run check` does not check for html code.
There is also some inconsistencies in `options_page.html` line 62 where a `</li>` is not in a newline.
Is this intended because it is the last `li` element?

Fixed the dynamic regex setting with the last commit.